app/services/region_report_service.rb

```diff
@@ -42,24 +43,40 @@ def compile_control_and_registration_data
   # the previous three quarters. Each quarter cohort is made up of patients registered
   # in the previous quarter who has had a follow up visit in the current quarter.
   def compile_cohort_trend_data
-    Quarter.new(date: selected_date).downto(3).each do |results_quarter|
-      cohort_quarter = results_quarter.previous_quarter
+    Rails.cache.fetch(cohort_cache_key, version: cohort_cache_version, expires_in: 7.days, force: force_cache?) do
```

Comment:
If my [understanding](https://github.com/simpledotorg/simple-server/pull/1149#issuecomment-658212422) of `cache_version` is correct,

Recyclable cache keys are valuable if the model is constantly _touched_, which is a typical CRUD case for Rails; Facilities almost never get updated; which means this will almost always cache for 7 days.

7 days feels like a long time to cache (even monthly) reports since it's not _that_ much data to show, we can probably afford to be more aggressive since these are just aggregates, so it's not in the order of of 100s of MBs (correct me if I'm wrong). Also since the matviews backing this refresh daily anyway, caching these for 7 days isn't that helpful, we could simply just cache for 2 days to match up with the worst-case scenario of view refresh to report cache refresh time.

Let's check with @dburka or @timcheadle on whether it's ok to do this more frequently?

Reply:
> Let's check with @dburka or @timcheadle on whether it's ok to do this more frequently?

Sure, we can change that expire time to whatever we want to after shipping this. We can also force whole districts from the web via the `force_cache` mechanism.

Reply:
The only problem with shortening the time is that that means some user will hit the expiration case every 2 days and experience a slow page load first time in.  We could prefill the cache to avoid that if we wanted to.